File: solutions/javascript/triangle/1/triangle.js
//
// This is only a SKELETON file for the 'Triangle' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

export class Triangle {
  constructor(...sides) {
    const [a, b, c] = sides;
    this.a = a;
    this.b = b;
    this.c = c;
  }

   isTriangle(){
    return (this.a+this.b)>=this.c && (this.b+this.c)>=this.a && (this.a+this.c)>=this.b &&
      this.a*this.b*this.c !== 0;
  }
  
  get isEquilateral() {
    if (this.isTriangle()){
      return this.a === this.b && this.b === this.c;
    }
    return false;
  }

  get isIsosceles() {
    if (this.isTriangle()){
      return this.a === this.b || this.b === this.c || this.a === this.c;
    }
    return false;
  }

  get isScalene() {
    if (this.isTriangle()) {
      return !this.isIsosceles;
    }
    return false;
  }
}

File: solutions/python/triangle/1/triangle.py
def istriangle(sides):
    a, b, c = sides
    if a>0 and b>0 and c>0:
        return a+b>=c and b+c>=a and a+c>=b
    return False

def equilateral(sides):
    if istriangle(sides):
        a, b, c = sides
        return a==b==c
    return False


def isosceles(sides):
    if istriangle(sides):
        a, b, c = sides
        return a==b!=c or a==c!=b or b==c!=a or a==b==c
    return False


def scalene(sides):
    if istriangle(sides):
        a, b, c = sides
        return a!=b!=c!=a
    return False
